client/components/checkout-form.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+export default class CheckoutForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      creditCard: null,
+      shippingAddress: ''
+    };
+    this.setView = this.setView.bind(this);
+    this.handleChangeName = this.handleChangeName.bind(this);
+    this.handleChangeCC = this.handleChangeCC.bind(this);
+    this.handleChangeAddress = this.handleChangeAddress.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+
+  handleChangeName(event) {
+    this.setState({ name: event.target.value });
+  }
+
+  handleChangeCC(event) {
+    this.setState({ creditCard: event.target.value });
+
+  }
+
+  handleChangeAddress(event) {
+    this.setState({ shippingAddress: event.target.value });
+  }
+
+  handleSubmit(event) {
+
+    event.preventDefault();
+
+    const order = {
+      name: this.state.name,
+      creditCard: this.state.creditCard,
+      shippingAddress: this.state.shippingAddress
+    };
+    this.props.order(order);
+  }
+
+  setView(e) {
+    this.props.view('catalog', {});
+  }
+
+  render() {
+
+    const totalPrice = (this.props.totalCost / 100).toFixed(2);
+
+    return (
+      <>
+        <div className="container">
+          <h2>My Cart</h2>
+          <p className="text-muted">{`Order Total: $${totalPrice}`}</p>
+          <form onSubmit={this.handleSubmit}>
+            <div className="form-group">
+              <label htmlFor="name">Name</label>
+              <input type="name" className="form-control" id="name" value={this.state.name} onChange={this.handleChangeName}/>
+            </div>
+
+            <div className="form-group">
+              <label htmlFor="cc">Credit Card Number:</label>
+              <input type="number" className="form-control" id="cc" value={this.state.creditCard} onChange={this.handleChangeCC}/>
```

Comment:
For the 2 inputs, the `type` attribute should be `text` then the `name` attribute should be: `name`, `number` respectively.